services/core-web/src/interfaces/actionCreator.type.ts

```diff
@@ -0,0 +1,17 @@
+import { AppThunk } from "../store/appThunk.type";
+
+/**
+
+ A type that takes a function type T as input and returns a new type based on the output type R.
+ If R is a type that extends AppThunk with a state of type S, then the ACType type will be a function that takes the same arguments as T, and returns a state of type S.
+ Otherwise, the ACType type will be undefined.
+ @typeparam T - A function type with any number and type of arguments.
+ @typeparam R - A result type.
+ @typeparam S - A state type.
+ @return - Either a function that takes the same arguments as T, and returns a state of type S, or undefined.
+ */
+export type ACType<T extends (...args: any[]) => any> = T extends (...args: infer P) => infer R
```

Comment:
<3 Generics! Thoughts on calling this `ActionCreator` instead of ACType to be a bit more descriptive? Also don't think there's a need to add the `Type` postfix - it's usually standard to omit that

Reply:
Good thought!  Yes, I like that naming better!